operator/redisfailover/service/generator.go

```diff
@@ -281,6 +281,196 @@ func generateHAProxyService(rf *redisfailoverv1.RedisFailover, labels map[string
 	}
 }
 
+func generateHAProxyRedisSlaveDeployment(rf *redisfailoverv1.RedisFailover, labels map[string]string, ownerRefs []metav1.OwnerReference) *appsv1.Deployment {
+
+	name := rf.GenerateName(redisHAProxySlaveRedisName)
+
+	namespace := rf.Namespace
+
+	labels = util.MergeLabels(labels, map[string]string{
+		"app.kubernetes.io/component": "redis",
```

Comment:
The headless Redis Headless Service is picking up HAProxy pods in addition to Redis pods. I think it's because of this component label. Is this intentional? I thought the Headless service was meant to point ONLY at Redis pods. 

https://github.com/powerhome/redis-operator/blob/b36b764b75f34742bec73066455ecd63d8c4373f/operator/redisfailover/service/generator.go#L213-L215

Perhaps the headless service should be using the `redisfailovers.databases.spotahome.com/component: redis` selector label instead to target only the redis pods of a RedisFailover? Or these haproxy pods should be labeled as `"app.kubernetes.io/component": "redis"`? The former feels less-wrong to me, but curious if you have any thoughts on this.

Reply:
My thoughts is that we need to change `app.kubernetes.io/component` for HAProxy someday. Probably new issue should be opened